sdk/ai/azure-ai-inference/src/main/java/com/azure/ai/inference/ChatCompletionClientTracer.java

```diff
@@ -0,0 +1,606 @@
+         * @param requestOptions The requestOptions parameter for the {@code operation}.
+         * @return chat completions for the provided chat messages.
+         */
+        Mono<ChatCompletions> invoke(BinaryData completeRequest, RequestOptions requestOptions);
+    }
+
+    /**
+     * Reference to the async operation performing the actual completion streaming call.
+     */
+    @FunctionalInterface
+    public interface StreamingCompleteOperation {
+        /**
+         * invokes the operation.
+         *
+         * @param completeRequest The completeRequest parameter for the {@code operation}.
+         * @param requestOptions The requestOptions parameter for the {@code operation}.
+         * @return chat completions streaming for the provided chat messages.
+         */
+        Flux<StreamingChatCompletionsUpdate> invoke(BinaryData completeRequest, RequestOptions requestOptions);
+    }
+    //</editor-fold>
+
+    /**
+     * Creates ChatCompletionClientTracer.
+     *
+     * @param endpoint the service endpoint.
+     */
+    ChatCompletionClientTracer(String endpoint) {
+        this(endpoint, isContentCapturingEnabled(), null);
+    }
+
+    /**
+     * Creates ChatCompletionClientTracer.
+     *
+     * @param endpoint the service endpoint.
+     * @param captureContent if full content should be captured.
+     * @param clientOptions the client options.
+     */
+    ChatCompletionClientTracer(String endpoint, boolean captureContent, ClientOptions clientOptions) {
+        this.logger = new ClientLogger(ChatCompletionClientTracer.class);
+        final URL url = parse(endpoint, logger);
+        if (url != null) {
+            this.host = url.getHost();
+            this.port = url.getPort() == -1 ? url.getDefaultPort() : url.getPort();
+        } else {
+            this.host = null;
+            this.port = -1;
+        }
+        this.captureContent = captureContent;
+        this.tracer = createTracer(clientOptions);
+    }
+
+    /**
+     * Traces the synchronous convenience API - {@link com.azure.ai.inference.ChatCompletionsClient#complete(ChatCompletionsOptions)}.
+     *
+     * @param request input options containing chat options for complete API.
+     * @param operation the operation performing the actual completion call.
+     * @param completeRequest The completeRequest parameter for the {@code operation}.
+     * @param requestOptions The requestOptions parameter for the {@code operation}.
+     * @return chat completions for the provided chat messages.
+     */
+    @SuppressWarnings("try")
+    ChatCompletions traceSyncComplete(ChatCompletionsOptions request, SyncCompleteOperation operation,
+        BinaryData completeRequest, RequestOptions requestOptions) {
+        if (!tracer.isEnabled()) {
+            return operation.invoke(completeRequest, requestOptions);
+        }
+        final Context span = tracer.start(rootSpanName(request), new StartSpanOptions(SpanKind.CLIENT), Context.NONE);
```

Comment:
we should take context as input parameter and use it as a parent to this span - this way someone may pass context explicitly 

Here's the use-case example in https://github.com/Azure/azure-sdk-for-java/blob/8250f3e42ee2af4ca8e842f47f7c582e454036bf/sdk/core/azure-core-tracing-opentelemetry-samples/src/samples/java/com/azure/core/tracing/opentelemetry/samples/TracingJavaDocCodeSnippets.java#L107

Reply:
Good catch, update to use context if available